assets/js/app.js

```diff
@@ -0,0 +1,51 @@
+const productName = document.getElementById('productName');
+const productPrice = document.getElementById('productPrice');
+const productType = document.getElementById('productType');
+const productImage = document.getElementById('productImage');
+const productForm = document.getElementById('productForm');
+const cartTableBody = document.querySelector('#cartTable tbody');
+
+let products = [];
+
+function addProduct(event) {
+    event.preventDefault();
+    const name = productName.value.trim();
+    const price = parseFloat(productPrice.value.trim());
+    const type = productType.value;
+    const imageUrl = productImage.value.trim();
+
+    if (!name || isNaN(price) || !type || !imageUrl) {
+        alert('All fields are required!');
+        return;
+    }
+
+    const product = { name, price, type, imageUrl };
+    products.push(product);
+
+    updateCartTable();
+    productForm.reset();
+}
+
+function updateCartTable() {
+    cartTableBody.innerHTML = '';
+
+    products.forEach((product, index) => {
+        const row = document.createElement('tr');
+        row.innerHTML = `
+            <td>${product.name}</td>
+            <td>${product.price.toFixed(2)}</td>
+            <td>${product.type}</td>
+            <td><img src="${product.imageUrl}" alt="${product.name}" style="max-width: 100px;"></td>
+            <td><button class="deleteButton" data-index="${index}">Delete</button></td>
+        `;
+        cartTableBody.appendChild(row);
```

Comment:
you manipulate the DOM in a loop which is bad practice. the browser will re-render every iteration. better to cumulate all "rows" in a string and then set them in a single command to the DOM

Reply:
על שניי ההערות האחרונות:שילוב של גוגל וchatgpt.
 כשהגעתי לכתוב את הפונקציה שמעדכנת את הטבלה הייתי על 30 אחוז והגיט commit לא עבד לי, אז ביקשתי מצאט שיכתוב את הקוד. 
ביקשתי הסבר לקוד מהצאט, חיפשתי את הדברים שהוא עשה בגוגל כדיי לבדוק שאני מבין את מה שהוא עשה. ואז הוספתי את זה לקוד. אחריי זה הרצתי שוב בצאט כדיי לראות שגיאות או יעול לקוד ושם נכנס הqueryselector, והלולאה שבודקת index כדי שהdelete ידע איזה שורה הוא מוחק מהטבלה. 
זכור לי כי אמרת שמותר להשתמש בכלים שנתונים לנו באנטרנט במהלך המבחן לכן אני מקווה שזה בסדר. אני לא גאה בזה כי אישית אני לא אוהב ai לשימוש כזה. זה מוציא את הצורך בכישרון ופיתוח הmuscle memory. אני יושב שעות על השיעוריי בית בדיוק מהסיבות האלו. אך במקרה זה חששתי שלא אספיק. לכן ווידאתי קודם שאני מבין מה הצאט רשם בפונקציה שביקשתי ממנו לכתוב לפניי שאני מגיש.